sentry-android-core/src/main/java/io/sentry/android/core/InternalSentrySdk.java

```diff
@@ -252,6 +264,26 @@ private static void addTimeSpanToSerializedSpans(TimeSpan span, List<Map<String,
     spans.add(spanMap);
   }
 
+  private static void deleteCurrentSessionFile(final @NotNull SentryOptions options) {
```

Comment:
Hmm why do you need to delete the file manually here? I see we anyway delete the session file on session end [here](https://github.com/getsentry/sentry-java/blob/7ab32b6ab8bc72b973328a00840eb192b151b59f/sentry/src/main/java/io/sentry/cache/EnvelopeCache.java#L103).

Reply:
Hmm, when I don't explicitly delete it, it's not removed and it's read on the next app start. But I didn't know it should be deleted automatically.

Reply:
If I'm seeing this right, the envelope is only deleted when the session end is send as standalone envelope, but in the case of hybrids the session end will be send with the crash event in the same envelope.

Reply:
Just for this is the warning I get if not cleaning the session file
https://github.com/getsentry/sentry-java/blob/6de823e05405663927ff66a16cfc5ab645529cc1/sentry/src/main/java/io/sentry/PreviousSessionFinalizer.java#L74